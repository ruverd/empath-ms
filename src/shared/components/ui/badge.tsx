import { cva, type VariantProps } from "class-variance-authority";

import { cn } from "@/shared/lib/utils";

const badgeVariants = cva(
  "inline-flex items-center border-shades-violet-5 rounded-full border px-2.5 py-1 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "text-secondary-foreground bg-shades-violet-3",
        outline: "text-foreground bg-transparent",
      },
    },
    defaultVariants: {
      variant: "default",
    },
  }
);

export interface BadgeProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof badgeVariants> {}

function Badge({ className, variant, ...props }: BadgeProps) {
  return (
    <div className={cn(badgeVariants({ variant }), className)} {...props} />
  );
}

export { Badge, badgeVariants };
