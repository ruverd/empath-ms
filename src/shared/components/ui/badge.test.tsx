import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { Badge } from "./badge";

describe("Badge", () => {
  it("should render with default variant", () => {
    render(<Badge>Test Badge</Badge>);

    const badge = screen.getByText("Test Badge");
    expect(badge).toBeInTheDocument();
  });

  it("should render with outline variant", () => {
    render(<Badge variant="outline">Test Badge</Badge>);

    const badge = screen.getByText("Test Badge");
    expect(badge).toBeInTheDocument();
    expect(badge).toHaveClass("text-foreground bg-transparent");
  });

  it("should apply custom className", () => {
    const customClass = "custom-class";
    render(<Badge className={customClass}>Test Badge</Badge>);

    const badge = screen.getByText("Test Badge");
    expect(badge).toHaveClass(customClass);
  });

  it("should render with all base classes", () => {
    render(<Badge>Test Badge</Badge>);

    const badge = screen.getByText("Test Badge");
    expect(badge).toHaveClass(
      "inline-flex",
      "items-center",
      "border-shades-violet-5",
      "rounded-full",
      "border",
      "px-2.5",
      "py-1",
      "text-xs",
      "font-semibold"
    );
  });

  it("should pass through additional props", () => {
    const testId = "test-badge";
    render(<Badge data-testid={testId}>Test Badge</Badge>);

    expect(screen.getByTestId(testId)).toBeInTheDocument();
  });

  it("should match snapshot with default variant", () => {
    const { container } = render(<Badge>Test Badge</Badge>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should match snapshot with outline variant", () => {
    const { container } = render(<Badge variant="outline">Test Badge</Badge>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
