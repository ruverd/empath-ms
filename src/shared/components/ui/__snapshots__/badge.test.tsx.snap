// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > should match snapshot with default variant 1`] = `
<div
  class="inline-flex items-center border-shades-violet-5 rounded-full border px-2.5 py-1 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-primary-cloud bg-shades-violet-3"
>
  Test Badge
</div>
`;

exports[`Badge > should match snapshot with outline variant 1`] = `
<div
  class="inline-flex items-center border-shades-violet-5 rounded-full border px-2.5 py-1 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground bg-transparent"
>
  Test Badge
</div>
`;
